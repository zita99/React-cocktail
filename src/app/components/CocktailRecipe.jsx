import React from 'react';

const CocktailRecipe = (props) => {

  const ingredientsList = [];
  for (let i = 1; i <= 15; i++) {
    if (props.recipe['strIngredient' + i]) {
      const ingredient = <li key={i}>{`${props.recipe['strIngredient' + i]} (${props.recipe['strMeasure' + i]})`}</li>;
      ingredientsList.push(ingredient);
    }
  }

  return (
      <div className='row recipes-list_element'>        
        <div className='recipes-list_main'>
          <h4>{ props.recipe.strDrink } </h4>
          <p>Category: { props.recipe.strCategory }</p>
          <p>Type: { props.recipe.strAlcoholic }</p>
          <div className='recipes-list_image'>
            <img src={ props.recipe.strDrinkThumb } alt='cocktail image small' onClick={ () => props.showFullImage(props.recipe.strDrinkThumb, props.recipe) }></img>
          </div>
        </div>
      </div>
      
  );
};

module.exports = CocktailRecipe;
