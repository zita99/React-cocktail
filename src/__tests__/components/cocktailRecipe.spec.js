import React from 'react';
import CocktailRecipe from '../../app/components/CocktailRecipe.jsx';
import { shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('CocktailRecipe', () => {

  it('should render if recipe object is passed', () => {

    const recipe = {
      strDrink: 'test',
      strCategory: 'testCategory',
      strAlcoholic: 'testAlcoholic',
      strDrinkThumb: 'testDrinkThumb',
      strInstructions: 'testInstructions'
    };

    const wrapper = shallow(
      <CocktailRecipe recipe={ recipe }/>
    );

    expect(toJson(wrapper)).toMatchSnapshot();

  });

});
