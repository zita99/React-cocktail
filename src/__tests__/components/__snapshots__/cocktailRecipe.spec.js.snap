// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CocktailRecipe should render if recipe object is passed 1`] = `
<div
  className="row recipes-list_element"
>
  <div
    className="col-6 recipes-list_main"
  >
    <h4>
      test
       
    </h4>
    <p>
      Category: 
      testCategory
    </p>
    <p>
      Type: 
      testAlcoholic
    </p>
    <div
      className="recipes-list_image"
    >
      <img
        alt="cocktail image small"
        onClick={[Function]}
        src="testDrinkThumb"
      />
    </div>
    <p>
      [click image to enlarge]
    </p>
  </div>
  <div
    className="col-6 recipes-list_ingredients"
  >
    <p
      className="recipes-list_ingredients_heading"
    >
      Ingredients:
    </p>
    <ul />
    <p
      className="recipes-list_ingredients_heading"
    >
      Instructions:
    </p>
    <p
      className="recipes-list_ingredients_instructions"
    >
      testInstructions
    </p>
  </div>
</div>
`;
